s3-private/s3.tf

```diff
@@ -7,8 +7,8 @@ module "s3" {
 
   versioning                  = var.versioning
   kms_deletion_window_in_days = var.kms_deletion_window_in_days
-  enable_encryption           = true
   multi_region_kms_key        = var.multi_region_kms_key
+  enable_encryption           = var.sse_algorithm == "aws:kms"
```

Comment:
Should this be `var.sse_algorithm != null`? Why would we not keep enforcing this at "true"?

Reply:
`enable_encryption` determine whether the KMS key is to be created. 
Set it to false here because we don't need KMS key if the sse_algorithm is AES256

Reply:
Thanks, that makes sense. Maybe `enable_encryption` should be named sth. like `with_custom_kms_key`?

Reply:
Will open an issue for this -> https://github.com/dbl-works/terraform/issues/295